File: PHP/5-DIP/DIP-Solucao/Cliente.php
<?php
require_once('Interfaces/ICPFServices.php');
require_once('Interfaces/IEmailServices.php');

class Cliente
{
    protected $cpfServices;
    protected $emailServices;

    public function __construct(
        ICPFServices $cpfServices, 
        IEmailServices $emailServices)
    {
        $this->cpfServices = $cpfServices;
        $this->emailServices = $emailServices;
    }

    public $clienteId;
    public $nome;
    public $email;
    public $cpf;
    public $dataCadastro;

    public function isValid()
    {
        return $this->emailServices->isValid($this->email) && $this->cpfServices->isValid($this->cpf);
    }
}

File: Python/5-DIP/DIP-Solucao/Cliente.py
import sys
from .interfaces.ICPFService import ICPFService 
from .interfaces.IEmailService import IEmailService 

class Cliente:
    def __init__(self, _cpfService: ICPFService, _emailService: IEmailService):
        self._cpfService = _cpfService
        self._emailService = _emailService

    clienteId: int
    nome: str
    email: str
    cpf: str
    dataCadastro: str

    def isValid(self):
        return self._emailService.isValid(self.email) and self._cpfService.isValid(self.cpf)
        